pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -1979,11 +1981,27 @@ impl Timeline {
                 info!("no layers to compact with gc: no historic layers below gc_cutoff, gc_cutoff={}", gc_cutoff);
                 return Ok(());
             };
+            let Some(min_layer_lsn) = layers
+                .iter_historic_layers()
+                .filter(|desc| {
+                    if let Some(compact_above_lsn) = options.compact_above_lsn {
+                        desc.get_lsn_range().end > compact_above_lsn // strictly larger than compact_above_lsn
```

Comment:
A few lines above, there's a comment `// Pick all the layers`.
It should be extended to cover the `compact_above_lsn`.

Reply:
https://github.com/neondatabase/neon/pull/9900/commits/563cdddd6a37e8e94d36c885274e0179a89984bf